optimum/onnxruntime/modeling_ort.py

```diff
@@ -445,18 +450,40 @@ def forward(
         token_type_ids: Optional[torch.Tensor] = None,
         **kwargs,
     ):
-        # converts pytorch inputs into numpy inputs for onnx
-        onnx_inputs = {
-            "input_ids": input_ids.cpu().detach().numpy(),
-            "attention_mask": attention_mask.cpu().detach().numpy(),
-        }
-        if token_type_ids is not None:
-            onnx_inputs["token_type_ids"] = token_type_ids.cpu().detach().numpy()
-        # run inference
-        outputs = self.model.run(None, onnx_inputs)
-        last_hidden_state = torch.from_numpy(outputs[self.model_outputs["last_hidden_state"]]).to(self.device)
-        # converts output to namedtuple for pipelines post-processing
-        return BaseModelOutput(last_hidden_state=last_hidden_state)
+        if self.device.type == "cuda" and self.use_io_binding:
+            onnx_inputs = {"input_ids": input_ids, "attention_mask": attention_mask}
+            if token_type_ids is not None:
+                onnx_inputs["token_type_ids"] = token_type_ids
+            io_helper = IOBindingHelper(self.model, self.config, self.device)
+            io_binding = io_helper.prepare_io_binding(**onnx_inputs)
```

Comment:
Will check the penalty! Indeed for models with fixed inputs and outputs we can do that, then there will be much more lines of code also...

Reply:
Hi @philschmid, just ran a test (batch size:3000, iteration: 100)
* Bind with IO helper: {'Inference_runtime': 112.6225}
* Direct binding in the ort model: {'Inference_runtime': 86.8001}

Ok, there is a significant penalty, let's move the preparation of binding to each ort model. 
(And maybe also leave the dynamic method in helper to leave a door for dynamic cases like custom task ort model)

Reply:
Thanks for the heads up!!  @philschmid

Reply:
> (And maybe also leave the dynamic method in helper to leave a door for dynamic cases like custom task ort model)

Do you think it is better to have it separate or in the CustomOrtModel. I think we can add it only the custom model since there are no other "users" of it right?

Reply:
Sure, moved it to the ort model for custom tasks.